### PR TITLE
chore: save encrypted password to authfile

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
saves encrypted password to the authfile

### What issues does this PR fix or reference?
@W-8776193@

### CODE REVIEW

File: src/commands/force/user/password/generate.ts [hunk 1]
```diff
@@ -55,9 +56,12 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
         // userId is used by `assignPassword` so we need to set it here
         authInfo.getFields().userId = fields.id;
         await user.assignPassword(authInfo, password);
+        const crypto = await Crypto.create();
```
Comment (amphro) [inline src/commands/force/user/password/generate.ts:59]: Do not use crypto directly.

```
password.value((pass) => {
    this.passwordData.push({ username: aliasOrUsername, password: pass.toString('utf-8') });
    authInfo.update(pass.toString('utf-8'));
  });
  await authInfo.save();
```

**Review by amphro — CHANGES_REQUESTED** [review]

File: src/commands/force/user/password/generate.ts (hunk 1 above)
Comment (amphro) [reply src/commands/force/user/password/generate.ts:59]: Something like that. The `update` should encrypt for you though.

**Review by amphro — APPROVED** [review]

**Review by shetzel — APPROVED** [review]

File: src/commands/force/user/list.ts [hunk 2]
```diff
@@ -38,14 +38,16 @@ export class UserListCommand extends SfdxCommand {
...
     const authList: AuthList[] = userAuthData.map((authData) => {
       const username = authData.getUsername();
-
+      // if they passed in a alias and it maps to something we have an Alias.
+      const alias = aliases.getKeysByValue(authData.getUsername())[0];
+      const configAgg = ConfigAggregator.getInstance();
+      const defaultUser = configAgg.getLocalConfig().get('defaultusername');
       return {
-        defaultMarker: authData.getFields().scratchAdminUsername ? '' : '(A)',
+        defaultMarker: defaultUser === username ? '(A)' : '',
```
Comment (amphro) [inline src/commands/force/user/list.ts:50]: Should this be `defaultUser === username || defaultUser === alias`?

File: test/commands/user/list.test.ts [hunk 3]
```diff
@@ -77,7 +77,7 @@ describe('force:user:list', () => {
       // testUser1@test.com is aliased to testUser
       const expected = [
         {
-          defaultMarker: '(A)',
```
Comment (amphro) [inline test/commands/user/list.test.ts:80]: It seems like there are a lot more code changes for the number of tests. How come these regressions weren't found by the unit tests originally? Can we modify the test a bit to make sure it is caught in the future?

**Pull request merged**